src/systems/pilot.ts

```diff
@@ -3,13 +3,33 @@ import { glMatrix, vec2 } from 'gl-matrix'
 import { TILE_SIZE } from '~/constants'
 import { make } from '~/entities/pilot'
 import { Game } from '~/Game'
+import { PathFinder } from '~/map/PathFinder'
 import { MouseButton } from '~/Mouse'
 
 export const update = (g: Game, dt: number): void => {
   // spawn pilot
   if (g.mouse.isUp(MouseButton.RIGHT) && g.mouse.getPos().isSome()) {
     const destPos = g.camera.viewToWorldspace(g.mouse.getPos().unwrap())
-    const pilot = make(destPos)
+
+    // Find tile positions of pilot and destination
+    const playerTilePos = vec2.floor(
+      vec2.create(),
+      vec2.scale(
+        vec2.create(),
+        g.player.unwrap().transform!.position,
+        1 / TILE_SIZE,
+      ),
+    )
+    const tilePos = vec2.floor(
+      vec2.create(),
+      vec2.scale(vec2.create(), destPos, 1 / TILE_SIZE),
+    )
+
+    // Generate path to tile position
+    const pathfinder = new PathFinder(g.map)
+    const path = pathfinder.discover(playerTilePos, tilePos)
```

Comment:
Pathfinding during spawning is a good optimization. I have been wondering whether we eventually have to worry about pathfinding every frame (or every few frames) if the target moves (e.g., the pilot returning to the tank), or obstacles update dynamically (e.g., walls are destroyed). Maybe there is a middle ground where we keep a list of all active paths, and re-path them on certain game events.